packages/amplify-graphql-auth-transformer/src/resolvers/query.ts

```diff
@@ -105,7 +105,7 @@ const generateAuthOnRelationalModelQueryExpression = (
               iff(
                 and([not(ref(IS_AUTHORIZED_FLAG)), methodCall(ref('util.isNull'), ref('ctx.stash.authFilter'))]),
                 compoundExpression([
-                  qref(methodCall(ref(`ctx.${claim}.put`), str(field), ref(`primaryRole${idx}`))),
+                  qref(methodCall(ref(`ctx.stash.sharedVariables.put`), str(field), ref(`primaryRole${idx}`))),
```

Comment:
Instead of `sharedVariables`, we can use very specific name like `connectionVariables` or `authVariables`.